src/main/scala/ai/privado/audit/AuditReportEntryPoint.scala

```diff
@@ -107,7 +105,7 @@ object AuditReportEntryPoint {
   ): Workbook = {
     val workbook: Workbook = new XSSFWorkbook()
     // Set Element Discovery Data into Sheet
-    val dataElementDiscoveryData = DataElementDiscovery.processDataElementDiscovery(xtocpg, taggerCache)
+    val dataElementDiscoveryData = DataElementDiscoveryJava.processDataElementDiscovery(xtocpg, taggerCache)
```

Comment:
I presume DataElementDiscovery is the same for every language with minor differences, why are we creating a `DataElementDiscoveryJava` object?

Reply:
Suggestion would be to have a single class which should be instantiated.

Reply:
No for Java/Kotlin we written some specific logic to detect the files within cpg.typeDecl
Additionally it has Java/kotlin specific filtering logic implemented.
Also, some specific logic for derivedClass which is not applicable for other languages.

So, let it be separate for now.